conductor/src/metrics_reporter.rs

```diff
@@ -80,45 +83,73 @@ pub async fn run_metrics_reporter() -> Result<()> {
                 info!("Enqueued batch {}/{} to PGMQ", i, batches);
                 i += 1;
             }
+            info!("Processed metric in {:?}", now.elapsed());
         }
     }
 }
 
 struct Client {
-    query_url: String,
+    prometheus_url: String,
+    loki_url: String,
     client: reqwest::Client,
 }
 
+#[derive(Debug, Deserialize)]
+#[serde(rename_all = "lowercase")]
+enum ServerType {
+    Prometheus,
+    Loki,
+}
+
+impl std::fmt::Display for ServerType {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            ServerType::Prometheus => write!(f, "Prometheus"),
+            ServerType::Loki => write!(f, "Loki"),
+        }
+    }
+}
+
 impl Client {
-    pub fn new() -> Self {
+    pub async fn new() -> Self {
         let prometheus_url = from_env_default(
             "PROMETHEUS_URL",
-            "http://monitoring-kube-prometheus-prometheus.monitoring.svc.cluster.local:9090",
+            "http://monitoring-kube-prometheus-prometheus.monitoring.svc.cluster.local:9090/api/v1/query",
+        );
+        let loki_url = from_env_default(
+            "LOKI_URL",
+            "http://loki-gateway.monitoring.svc.cluster.local/loki/api/v1/query",
         );
-        let query_url = format!("{prometheus_url}/api/v1/query");
 
-        info!("metrics_reporter will use '{query_url}'");
+        info!("metrics_reporter will use '{prometheus_url}' for Prometheus");
+        info!("metrics_reporter will use '{loki_url}' for Loki");
 
         Self {
-            query_url,
+            prometheus_url,
+            loki_url,
             client: reqwest::Client::new(),
         }
     }
 
-    pub async fn query(&self, query: &str) -> Result<Metrics> {
-        let response = self
-            .client
-            .get(&self.query_url)
-            .query(&[("query", query)])
-            .send()
-            .await?;
+    pub async fn query(&self, query: &str, server_type: &ServerType) -> Result<Metrics> {
+        let query_url = match server_type {
+            ServerType::Prometheus => &self.prometheus_url,
+            ServerType::Loki => &self.loki_url,
+        };
+
+        let mut request = self.client.get(query_url).query(&[("query", query)]);
```

Comment:
I should add a timeout, and also change the code so that any given metrics query having an issue does not cause all of the subsequent queries to fail to process

Reply:
Right now I think it won't stop subsequent queries, right? Will just log and move on

Reply:
It looks to me like a failed query will stop the loop
```
let metrics = client.query(&metric.query, &metric.server).await?;
```